src/Symfony/Component/Config/DependencyInjection/ConfigCachePass.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Config\DependencyInjection;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Reference;
+
+/**
+ * Adds tagged metadata_validator services to config_cache_factory service.
+ *
+ * @author Benjamin Klotz <bk@webfactory.de>
+ */
+class ConfigCachePass implements CompilerPassInterface
+{
+    public function process(ContainerBuilder $container)
+    {
+        $definition = $container->getDefinition('config_cache_factory');
+
+        foreach ($container->findTaggedServiceIds('config_cache.metadata_validator') as $id => $attributes) {
+            $definition->addMethodCall('addValidator', array(new Reference($id)));
```

Comment:
I think that there should not be more than one authoritative `MetadataValidator` per resource type and if there are more, they should not disagree about freshness. 

Also, in order to keep things simple, we decided against coming up with a full-fledged voting strategy like in the Security component.